theano/gof/type.py

```diff
@@ -795,3 +796,137 @@ def signature(self):
         return (self.type,)
 
 CDataType.Constant = CDataTypeConstant
+
+
+class EnumType(Type, dict):
```

Comment:
What about

```python
params_type  = CEnumType('cudnnSoftmaxAlgorithm_t', ('CUDNN_SOFTMAX_SOFT', 'CUDNN_SOFTMAX_ACCURATE'))
```

Then in python we have the simple ints that you propose.

And in the C++ code in c_extract() for CEnumType it does something like this:

```C
switch (PyInt_AsLong(py_val)) {
 case 0:
  val = CUDNN_SOFTMAX_SOFT;
  break;
 case 1:
  val = CUDNN_SOFTMAX_ACCURATE;
  break;
 default:
  %(fail)s
}
```